### PR TITLE
chore: use Node.js 16.x for building package and tests

### PR DESCRIPTION
**Pull request by ctavan**
Previously we were requiring Node 12.x to get somewhat reproducible npm builds. This now upgrades the requirement to the latests LTS version 16.x

### CODE REVIEW

**Review by TrySound — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,14 +31,14 @@ jobs:
...
-      if: matrix.node-version == '12.x'
+      if: matrix.node-version == '16.x'
       env:
         BUNDLEWATCH_GITHUB_TOKEN: ${{ secrets.BUNDLEWATCH_GITHUB_TOKEN }}
     - run: npm run test:node
-      if: matrix.node-version == '12.x' || matrix.node-version == '14.x'
+      if: matrix.node-version == '16.x' || matrix.node-version == '14.x'
     - run: npm run test:pack
-      if: matrix.node-version == '14.x'
+      if: matrix.node-version == '16.x' || matrix.node-version == '14.x'
```
Comment (LinusU) [inline .github/workflows/ci.yml:44]: I think that these two tests should run on `>= 12`? That is, all that supports ESM

**Comment by LinusU** [comment]
> npm ERR! code EINVALIDPACKAGENAME
> npm ERR! Invalid package name ".local": name cannot start with a period

Ran into this locally when using npm 7 as well, I didn't have time to look into why there was a `.local` folder though...

**Comment by ctavan** [comment]
> > npm ERR! code EINVALIDPACKAGENAME
> > npm ERR! Invalid package name ".local": name cannot start with a period
> 
> Ran into this locally when using npm 7 as well, I didn't have time to look into why there was a `.local` folder though. It's a little annoying as it makes all files show up twice when searching in VS Code 😅

The `.local` symlink is a workaround to make local file references work in the examples folder (so you can actually test local changes)… The error you mentioned should now be fixed. Not sure about vscode. Could it be added to some ignore file for the editor?

**Comment by ctavan** [comment]
I had to upgrade @wdio v6 -> v7 in order to get it to run with Node 16 and unfortunately that was a bit of a rabbit hole but it should be good again :).

I couldn't resist upgrading all other build deps as well.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ctavan) [reply .github/workflows/ci.yml]: Early versions of 12.x didn't support ESM IIRC… That's why historically we didn't test there.

But I guess nowadays it's safe to run these tests on Node 12+, so thanks for the catch!

File: .local/uuid/abc.js [hunk 2]
```diff
@@ -0,0 +1 @@
+sdfkl;
```
Comment (LinusU) [inline .local/uuid/abc.js:1]: Should this be committed?

Comment (ctavan) [reply .local/uuid/abc.js:1]: Oops, of course not…. I was Troubleshooting eslint/prettier… will remove

**Review by LinusU — APPROVED** [review]
Nice! 👍

File: wdio.conf.js [hunk 3]
```diff
@@ -7,94 +7,111 @@ const GITHUB_REF = process.env.GITHUB_REF || '';
...
-    browser_version: '11.0',
-    os: 'Windows',
-    os_version: '7',
+    browserVersion: '11.0',
   },
 
   // Chrome
   {
-    ...commonCapabilities,
+    'bstack:options': {
```
Comment (LinusU) [inline wdio.conf.js:39]: Latest `main` have another entry here, just adding this comment so we don't miss updating that when merging/rebasing. I think that the merge/rebase will apply cleanly, but there will be one entry not updated to the new format ☺️

**Review by LinusU — APPROVED** [review]
Neat! 🙌

**Review by broofa — APPROVED** [review]

**Pull request merged**